python/app/dialog.py

```diff
@@ -1142,3 +1206,278 @@ def setup_entity_model_view(self, entity_data):
+        self.sort_menu_button.setStyleSheet("border :None")
+
+        # show the sort menu when the button is clicked
+        self.sort_menu_button.clicked.connect(
+            lambda: self._entity_field_menu.exec_(QtGui.QCursor.pos())
+        )
+
+        # Set the sort menu icon
+        icon_path = self._switch_sort_icon()
+        self.sort_menu_button.setIcon(QtGui.QIcon(icon_path))
+
+        # Add the sort menu button The Sort Filter Horizontal Layout
+        self._sort_filter_layout.addWidget(self.sort_menu_button)
+
+        # Nest the sort and filter menus layout to the label_menus Horizontal Layout
+        self._label_menus_layout.addLayout(self._sort_filter_layout)
+        # Nest the label_menus Horizontal Layout to the Vertical main app layout
+        task_tab_data["widget"].layout().addLayout(self._label_menus_layout)
+
+        # the fields manager is used to query which fields are supported
+        # for display. it can also be used to find out which fields are
+        # visible to the user and editable by the user. the fields manager
+        # needs time to initialize itself. once that's done, the widgets can
+        # begin to be populated.
+        fields_manager = shotgun_fields.ShotgunFieldManager(
+            self, bg_task_manager=self._task_manager
+        )
+        fields_manager.initialized.connect(self._field_filters)
+        fields_manager.initialize()
+        self._sort_menu_actions()
+
+    def _field_filters(self):
+
+        # ---- define a few simple filter methods for use by the menu
+
+        def field_filter(field):
+            # none of the fields are included
+            return False
+
+        def checked_filter(field):
+            # none of the fields are checked
+            return False
+
+        def disabled_filter(field):
+            # none of the fields are disabled
+            return False
+
+        # attach the filters
+        self._entity_field_menu.set_field_filter(field_filter)
+        self._entity_field_menu.set_checked_filter(checked_filter)
+        self._entity_field_menu.set_disabled_filter(disabled_filter)
+
+    def _sort_menu_actions(self):
+        """
+        Populate the sort menu with actions.
+        """
+
+        # Create Sort Menu actions
+        sort_asc = self._entity_field_menu._get_qaction("ascending", "Ascending")
+        sort_desc = self._entity_field_menu._get_qaction("descending", "Descending")
+        separator = self._entity_field_menu.addSeparator()
+        status_action = self._entity_field_menu._get_qaction("sg_status_list", "Status")
+        step_action = self._entity_field_menu._get_qaction("step", "Step")
+        start_date_action = self._entity_field_menu._get_qaction(
+            "start_date", "Start date"
+        )
+        due_date_action = self._entity_field_menu._get_qaction("due_date", "Due date")
+        id_action = self._entity_field_menu._get_qaction("id", "Id")
```

Comment:
It would be nice if the list of sort actions was configurable via a hook. Also it looks like this list of sort actions is specific to Tasks only, ideally we could have the sort menu for any entity type.